components/datetime/src/fields/symbols.rs

```diff
@@ -416,3 +434,62 @@ impl From<DayPeriod> for FieldSymbol {
         Self::DayPeriod(input)
     }
 }
+
+#[derive(Debug, Eq, PartialEq, Clone, Copy)]
+#[cfg_attr(
+    feature = "provider_serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
+pub enum TimeZone {
+    LowerZ,
+    UpperZ,
+    UpperO,
+    LowerV,
+    UpperV,
+    LowerX,
+    UpperX,
+}
+
+impl LengthType for TimeZone {
+    fn get_length_type(&self, length: &FieldLength) -> TextOrNumeric {
+        use TextOrNumeric::*;
+        match self {
+            TimeZone::UpperZ => match u8::from(*length) {
+                1..=3 => Numeric,
+                4 => Text,
+                5 => Numeric,
+                _ => panic!("Eear
+                pected field of valid length."),
+            },
+            TimeZone::UpperO => match u8::from(*length) {
+                1 => Text,
+                4 => Numeric,
+                _ => panic!("Expected field of valid length."),
+            },
+            TimeZone::LowerZ | TimeZone::LowerV | TimeZone::UpperV => Text,
+            TimeZone::LowerX | TimeZone::UpperX => Numeric,
+        }
```

Comment:
Hey @gregtatum 

I just rebased on top of your latest merge and had to impl LengthType for the zone patterns. 

Can you give this a quick look and make sure it's correct?
https://unicode.org/reports/tr35/tr35-dates.html#dfst-zone

My biggest concern was formats like `GMT+09:00` which have text and numbers. I made the assumption that this resolved to `Text`.

Reply:
I don't think this matters particularly. This is the line where this matters in the skeleton matching. Here the symbol would mismatch every time as it's implemented. I think I would default to always returning Text. This designation is more important for fields such as "month". The time zone is more orthogonal to the rest of the matching.

```rust
distance += if requested_field == skeleton_field {
  NO_DISTANCE
} else if requested_field.symbol != skeleton_field.symbol {
  SUBSTANTIAL_DIFFERENCES_DISTANCE
} else if requested_field.get_length_type() != skeleton_field.get_length_type()
{
  TEXT_VS_NUMERIC_DISTANCE
} else {
  WIDTH_MISMATCH_DISTANCE
};
```

Reply:
I think the panics will go away if you take my above suggestion of always using Text. However, with the current code I would change it into a `debug_assert` and be more permissive for a release build.